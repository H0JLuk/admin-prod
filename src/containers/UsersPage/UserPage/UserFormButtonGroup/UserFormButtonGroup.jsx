import { Button } from 'antd';
import React from 'react';
import PropTypes from 'prop-types';

const BUTTON_DELETE_TEXT = 'Удалить';
const BUTTON_CANCEL_TEXT = 'Отменить';
const BUTTON_ADD_USER_TEXT = 'Добавить';
const BUTTON_SAVE_EDIT_USER_TEXT = 'Сохранить';
const INFO_USER_BUTTONS = {
    RESET_PASSWORD: 'Сбросить пароль',
    EDIT: 'Редактировать',
    UNBLOCK: 'Разблокировать',
};

const UserFormButtonGroup = ({
    type,
    onDelete,
    onCancel,
    onSubmit,
    onResetPassword,
    onEditUser,
    disableAllButtons,
    userBlocked,
}) => {
    const deleteButton = (
        <Button type="primary" danger onClick={ onDelete } disabled={ disableAllButtons }>
            { BUTTON_DELETE_TEXT }
        </Button>
    );

    const cancelButton = (
        <Button type="ghost" onClick={ onCancel } disabled={ disableAllButtons }>
            { BUTTON_CANCEL_TEXT }
        </Button>
    );

    switch (type) {
        case 'new': {
            return (
                <>
                    { cancelButton }
                    <Button type="primary" onClick={ onSubmit } disabled={ disableAllButtons }>
                        { BUTTON_ADD_USER_TEXT }
                    </Button>
                </>
            );
        }
        case 'edit': {
            return (
                <>
                    { cancelButton }
                    <Button type="primary" onClick={ onSubmit } disabled={ disableAllButtons }>
                        { BUTTON_SAVE_EDIT_USER_TEXT }
                    </Button>
                    { deleteButton }
                </>
            );
        }
        case 'info': {
            return (
                <>
                    <Button type="primary" onClick={ onResetPassword } disabled={ disableAllButtons }>
                        { userBlocked ? INFO_USER_BUTTONS.UNBLOCK : INFO_USER_BUTTONS.RESET_PASSWORD }
                    </Button>
                    <Button type="primary" onClick={ onEditUser } disabled={ disableAllButtons }>
                        { INFO_USER_BUTTONS.EDIT }
                    </Button>
                    { deleteButton }
                </>
            );
        }
        default: {
            return null;
        }
    }
};

UserFormButtonGroup.propTypes = {
    type: PropTypes.oneOf(['info', 'new', 'edit']),
    onDelete: PropTypes.func,
    onCancel: PropTypes.func,
    onSubmit: PropTypes.func,
    onResetPassword: PropTypes.func,
    onEditUser: PropTypes.func,
    disableAllButtons: PropTypes.bool,
    userBlocked: PropTypes.bool,
};

UserFormButtonGroup.defaultProps = {
    type: 'info',
    onDelete: () => {},
    onCancel: () => {},
    onSubmit: () => {},
    onResetPassword: () => {},
    onEditUser: () => {},
    disableAllButtons: false,
    userBlocked: false,
};

export default UserFormButtonGroup;
