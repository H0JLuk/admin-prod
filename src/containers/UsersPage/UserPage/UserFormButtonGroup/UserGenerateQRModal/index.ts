import UserGenerateQRModal from './UserGenerateQRModal';

export default UserGenerateQRModal;
