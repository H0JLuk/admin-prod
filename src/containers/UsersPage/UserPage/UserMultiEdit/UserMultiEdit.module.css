.container {
    color: #494949;
    padding: 20px;
    min-width: 500px;
}

.container h4 {
    color: #ababad;
    font-size: 12px;
}

.header {
    margin-bottom: 25px;
}

.pageTitle {
    font-size: 20px;
    font-weight: 600;
}

.formContainer {
    width: 100%;
    background-color: #f7f7f7;
    padding: 25px 25px 46px 25px ;
    margin-bottom: 15px;
}

.form {
    display: flex;
    align-items: baseline;
    max-width: 900px;
}

.form > div {
    flex-basis: 50% !important;
}

.formError {
    margin-top: 15px;
    color: #ec464e;
}

.buttonsContainer button:not(:last-child) {
    margin-right: 15px;
}

.buttonsContainer :global(.ant-btn[disabled]) {
    color: #ebe8e8 !important;
}

.select {
    display: flex;
    flex-direction: column;
    margin-left: 10px;
    min-width: 240px;
    max-width: 300px;
}

.select label {
    margin-bottom: 8px;
}
