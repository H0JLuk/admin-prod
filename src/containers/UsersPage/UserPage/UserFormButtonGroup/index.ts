import UserFormButtonGroup from './UserFormButtonGroup';

export { BUTTON } from './UserFormButtonGroup';

export default UserFormButtonGroup;
