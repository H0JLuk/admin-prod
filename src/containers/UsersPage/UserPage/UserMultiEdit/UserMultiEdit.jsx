import React, { useState, useCallback, useEffect } from 'react';
import { useHistory, useLocation } from 'react-router-dom';
import { Form } from 'antd';
import { editLocationAndSalePointUsers, removeUser } from '../../../../api/services/usersService';
import Header from '../../../../components/Header/Header';
import AutocompleteLocationAndSalePoint from '../../../../components/Form/AutocompleteLocationAndSalePoint/AutocompleteLocationAndSalePoint';
import UserFormButtonGroup from '../UserFormButtonGroup/UserFormButtonGroup';

import styles from './UserMultiEdit.module.css';

const PAGE_TITLE = 'Пользователи';
export const CHOSE_USERS_COUNT = 'Выбрано пользователей';

const LOCATION_FIELD = {
    label: 'Выберите локацию',
};
const SALE_POINT_FIELD = {
    label: 'Выберите точку продажи',
};
const layout = {
    labelCol: { span: 8 },
    wrapperCol: { span: 12 },
};

const ACTION_PERMISSIONS = {
    deleteUser: true,
};

const UserMultiEdit = ({ matchPath }) => {
    const [location, setLocation] = useState(null);
    const [salePoint, setSalePoint] = useState(null);
    const [error, setError] = useState({ location: '', salePoint: '', backend: '' });
    const [isSendingInfo, setIsSendingInfo] = useState(false);
    const [arrUsersIds, setArrUsersIds] = useState([]);
    const [userRoleExist, setUserRoleExist] = useState(false);
    const history = useHistory();
    const { state: { users = [] } = {} } = useLocation();

    const redirectToUsersPage = useCallback(() => history.push(matchPath), [history, matchPath]);

    useEffect(() => {
        if (!users?.length) {
            redirectToUsersPage();
        }

        setUserRoleExist(users.some(({ role }) => role === 'User'));
        setArrUsersIds(users.map(({ id }) => id));
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    const onLocationChange = useCallback((location) => {
        setLocation(location);
        setError({ location: '', salePoint: '', backend: '' });
    }, []);

    const onSalePointChange = useCallback((salePoint) => {
        setSalePoint(salePoint);
        setError({ location: '', salePoint: '', backend: '' });
    }, []);

    const onCancel = useCallback(() => redirectToUsersPage(), [redirectToUsersPage]);

    const onDelete = useCallback(async () => {
        setIsSendingInfo(true);

        // TODO: изменить этот кусок кода, когда на бэке добавят API для удаления нескольких пользователей
        try {
            const requestPromises = arrUsersIds.map(removeUser);

            await Promise.all(requestPromises);
            redirectToUsersPage();
        } catch (e) {
            setIsSendingInfo(false);
            setError({ location: '', salePoint: '', backend: e.message });
        }
    }, [redirectToUsersPage, arrUsersIds]);

    const onSubmit = useCallback(async () => {
        if (userRoleExist && !salePoint) {
            return setError({ location: '', salePoint: 'Выберите действительную точку продажи', backend: '' });
        }

        setIsSendingInfo(true);

        try {
            await editLocationAndSalePointUsers({
                userIds: arrUsersIds,
                salePointId: salePoint?.id,
            });
            redirectToUsersPage();
        } catch (e) {
            setIsSendingInfo(false);
            setError({ location: '', salePoint: '', backend: e.message });
        }
    }, [salePoint, arrUsersIds, redirectToUsersPage, userRoleExist]);

    return (
        <>
            <Header />
            <div className={ styles.container }>
                <div className={ styles.header }>
                    <h4>{ PAGE_TITLE }</h4>
                    <div className={ styles.pageTitle }>
                        { `${CHOSE_USERS_COUNT} ${arrUsersIds.length}` }
                    </div>
                </div>
                <div className={ styles.formContainer }>
                    <Form { ...layout } className={ styles.form } requiredMark={ false }>
                        <AutocompleteLocationAndSalePoint
                            layout={ layout }
                            locationDisabled={ isSendingInfo }
                            salePointDisabled={ isSendingInfo }
                            locationLabel={ LOCATION_FIELD.label }
                            salePointLabel={ SALE_POINT_FIELD.label }
                            salePointLabelClassNames={ userRoleExist ? 'required': '' }
                            onLocationChange={ onLocationChange }
                            onSalePointChange={ onSalePointChange }
                            autoFocusLocation={ true }
                            locationId={ location?.id }
                            error={ error }
                        />
                    </Form>
                    { !!error.backend && <div className={ styles.formError }>{ error.backend }</div> }
                </div>
                <div className={ styles.buttonsContainer }>
                    <UserFormButtonGroup
                        type="edit"
                        onCancel={ onCancel }
                        onSubmit={ onSubmit }
                        onDelete={ onDelete }
                        disableAllButtons={ isSendingInfo }
                        actionPermissions={ ACTION_PERMISSIONS }
                    />
                </div>
            </div>
        </>
    );
};

export default UserMultiEdit;
