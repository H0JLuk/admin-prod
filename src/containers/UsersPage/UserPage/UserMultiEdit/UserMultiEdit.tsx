import React, { useState, useEffect } from 'react';
import { useHistory, useLocation } from 'react-router-dom';
import { Form } from 'antd';

import { editLocationAndSalePointUsers, removeUser } from '@apiServices/usersService';
import { LocationDto, SalePointDto, UserInfo } from '@types';
import AutocompleteLocationAndSalePoint from '@components/AutoComplete/AutocompleteLocationAndSalePoint';
import Header from '@components/Header';
import UserFormButtonGroup from '../UserFormButtonGroup';
import { ROLES_FOR_EXTERNAL_SALE_POINT, validateSalePoint } from '../UserFormHelper';

import styles from './UserMultiEdit.module.css';

const PAGE_TITLE = 'Пользователи';
export const CHOSE_USERS_COUNT = 'Выбрано пользователей';

const LOCATION_FIELD = {
    label: 'Выберите локацию',
};
const SALE_POINT_FIELD = {
    label: 'Выберите точку продажи',
};
const layout = {
    labelCol: { span: 8 },
    wrapperCol: { span: 12 },
};

const ACTION_PERMISSIONS = {
    deleteUser: true,
};

type UserMultiEditProps = {
    matchPath: string;
};

type LocationState = {
    users: UserInfo[];
};

const UserMultiEdit: React.FC<UserMultiEditProps> = ({ matchPath }) => {
    const [location, setLocation] = useState<LocationDto | null>(null);
    const [salePoint, setSalePoint] = useState<SalePointDto | null>(null);
    const [error, setError] = useState({ location: '', salePoint: '', backend: '' });
    const [isSendingInfo, setIsSendingInfo] = useState(false);
    const [arrUsersIds, setArrUsersIds] = useState<number[]>([]);
    const history = useHistory();
    const { state: { users = [] } = {} } = useLocation<LocationState>();

    const redirectToUsersPage = () => history.push(matchPath);

    useEffect(() => {
        if (!users?.length) {
            redirectToUsersPage();
        }
        setArrUsersIds(users.map(({ id }) => id));
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    const onLocationChange = (selectedLocation: LocationDto | null) => {
        setLocation(selectedLocation);
        setError({ location: '', salePoint: '', backend: '' });
    };

    const onSalePointChange = (selectedSalePoint: SalePointDto | null) => {
        setSalePoint(selectedSalePoint);
        setError({ location: '', salePoint: '', backend: '' });
    };

    const onDelete = async () => {
        setIsSendingInfo(true);

        try {
            const requestPromises = arrUsersIds.map(removeUser);

            await Promise.all(requestPromises);
            redirectToUsersPage();
        } catch (e) {
            setIsSendingInfo(false);
            setError({ location: '', salePoint: '', backend: e.message });
        }
    };

    const onSubmit = async () => {
        try {
            const salePointShouldExternal = users.every(({ role }) => ROLES_FOR_EXTERNAL_SALE_POINT.includes(role));
            validateSalePoint(salePoint, salePointShouldExternal);

            setIsSendingInfo(true);

            try {
                await editLocationAndSalePointUsers({
                    userIds: arrUsersIds,
                    salePointId: salePoint?.id,
                });
                redirectToUsersPage();
            } catch (e) {
                setIsSendingInfo(false);
                setError({ location: '', salePoint: '', backend: e.message });
            }
        } catch ({ message, name }) {
            return setError({
                location: '',
                salePoint: '',
                backend: '',
                [name]: message,
            });
        }
    };

    return (
        <>
            <Header />
            <div className={styles.container}>
                <div className={styles.header}>
                    <h4>{PAGE_TITLE}</h4>
                    <div className={styles.pageTitle}>
                        {`${CHOSE_USERS_COUNT} ${arrUsersIds.length}`}
                    </div>
                </div>
                <div className={styles.formContainer}>
                    <Form {...layout} className={styles.form} requiredMark={false}>
                        <AutocompleteLocationAndSalePoint
                            locationDisabled={isSendingInfo}
                            salePointDisabled={isSendingInfo}
                            locationLabel={LOCATION_FIELD.label}
                            salePointLabel={SALE_POINT_FIELD.label}
                            salePointLabelClassNames="required"
                            onLocationChange={onLocationChange}
                            onSalePointChange={onSalePointChange}
                            autoFocusLocation={true}
                            locationId={location?.id}
                            error={error}
                        />
                    </Form>
                    {!!error.backend && <div className={styles.formError}>{error.backend}</div>}
                </div>
                <div className={styles.buttonsContainer}>
                    <UserFormButtonGroup
                        type="edit"
                        onCancel={redirectToUsersPage}
                        onSubmit={onSubmit}
                        onDelete={onDelete}
                        disableAllButtons={isSendingInfo}
                        actionPermissions={ACTION_PERMISSIONS}
                    />
                </div>
            </div>
        </>
    );
};

export default UserMultiEdit;
