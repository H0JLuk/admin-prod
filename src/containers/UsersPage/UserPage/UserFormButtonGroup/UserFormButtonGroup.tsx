import React from 'react';
import { Button } from 'antd';
import noop from 'lodash/noop';
import PropTypes from 'prop-types';
import UserGenerateQRModal from './UserGenerateQRModal';
import { LOGIN_TYPES_ENUM } from '@constants/loginTypes';
import { InteractionsCurrUserToOtherUser } from '@constants/permissions';
import { ClientAppDto, UserInfo } from '@types';
import { BUTTON_TEXT } from '@constants/common';

type UserFormButtonGroupProps = {
    type?: 'info' | 'new' | 'edit' | string;
    clientApps?: ClientAppDto[];
    onDelete?: () => void;
    onCancel?: () => void;
    onSubmit?: () => void;
    onResetPassword?: () => void;
    onEditUser?: () => void;
    disableAllButtons?: boolean;
    userData?: UserInfo | null;
    actionPermissions?: InteractionsCurrUserToOtherUser;
};

export const INFO_USER_BUTTONS = {
    RESET_PASSWORD: 'Сбросить пароль',
    EDIT: 'Редактировать',
    UNBLOCK: 'Разблокировать',
    QR_CODE_GEN_TEXT: 'Сгенерировать QR-код',
};

const UserFormButtonGroup: React.FC<UserFormButtonGroupProps> = ({
    type,
    clientApps,
    onDelete,
    onCancel,
    onSubmit,
    onResetPassword,
    onEditUser,
    disableAllButtons,
    userData,
    actionPermissions,
}) => {
    const { deleteUser, editUser, resetUserPassword, unlockUser, canGenerateQRCode } = actionPermissions || {};
    const { tmpBlocked, personalNumber, loginType, clientAppIds, id } = userData || {};

    const deleteButton = deleteUser && (
        <Button
            type="primary"
            onClick={onDelete}
            disabled={disableAllButtons}
            danger
        >
            {BUTTON_TEXT.DELETE}
        </Button>
    );

    const cancelButton = (
        <Button
            type="ghost"
            onClick={onCancel}
            disabled={disableAllButtons}
        >
            {BUTTON_TEXT.CANCEL}
        </Button>
    );

    switch (type) {
        case 'new': {
            return (
                <>
                    {cancelButton}
                    <Button
                        type="primary"
                        onClick={onSubmit}
                        disabled={disableAllButtons}
                    >
                        {BUTTON_TEXT.ADD}
                    </Button>
                </>
            );
        }
        case 'edit': {
            return (
                <>
                    {cancelButton}
                    <Button
                        type="primary"
                        onClick={onSubmit}
                        disabled={disableAllButtons}
                    >
                        {BUTTON_TEXT.SAVE}
                    </Button>
                    {deleteButton}
                </>
            );
        }
        case 'info': {
            const showResetOrUnlockBtn =
                loginType === LOGIN_TYPES_ENUM.PASSWORD &&
                (
                    (tmpBlocked && unlockUser) ||
                    (!tmpBlocked && resetUserPassword)
                );
            const showQRCodeBtn = loginType === LOGIN_TYPES_ENUM.DIRECT_LINK && canGenerateQRCode;
            return (
                <>
                    {showQRCodeBtn && (
                        <UserGenerateQRModal
                            clientApps={clientApps}
                            personalNumber={personalNumber}
                            userId={id}
                            userClientAppIds={clientAppIds}
                            buttonDisabled={disableAllButtons}
                        />
                    )}
                    {showResetOrUnlockBtn && (
                        <Button
                            type="primary"
                            onClick={onResetPassword}
                            disabled={disableAllButtons}
                        >
                            {tmpBlocked ? INFO_USER_BUTTONS.UNBLOCK : INFO_USER_BUTTONS.RESET_PASSWORD}
                        </Button>
                    )}
                    {editUser && (
                        <Button
                            type="primary"
                            onClick={onEditUser}
                            disabled={disableAllButtons}
                        >
                            {INFO_USER_BUTTONS.EDIT}
                        </Button>
                    )}
                    {deleteButton}
                </>
            );
        }
        default: {
            return null;
        }
    }
};

UserFormButtonGroup.propTypes = {
    type: PropTypes.oneOf(['info', 'new', 'edit']),
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    clientApps: PropTypes.array,
    onDelete: PropTypes.func,
    onCancel: PropTypes.func,
    onSubmit: PropTypes.func,
    onResetPassword: PropTypes.func,
    onEditUser: PropTypes.func,
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    userData: PropTypes.object,
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    actionPermissions: PropTypes.object,
    disableAllButtons: PropTypes.bool,
};

UserFormButtonGroup.defaultProps = {
    type: 'info',
    onDelete: noop,
    onCancel: noop,
    onSubmit: noop,
    onResetPassword: noop,
    onEditUser: noop,
    disableAllButtons: false,
};

export default UserFormButtonGroup;
