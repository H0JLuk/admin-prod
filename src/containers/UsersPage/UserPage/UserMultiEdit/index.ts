import UserMultiEdit from './UserMultiEdit';

export default UserMultiEdit;
