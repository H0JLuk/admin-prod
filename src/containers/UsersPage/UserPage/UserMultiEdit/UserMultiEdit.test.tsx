/* eslint-disable @typescript-eslint/no-non-null-assertion */
import React from 'react';
import { mount, shallow } from 'enzyme';
import UserMultiEdit, { CHOSE_USERS_COUNT } from './UserMultiEdit';
import { editLocationAndSalePointUsers, removeUser } from '@apiServices/usersService';
import { sleep } from '@setupTests';
import { act } from 'react-dom/test-utils';
import { testSalePoint, testLocation, usersTestArray } from '@testConstants';
import AutocompleteLocationAndSalePoint from '@components/AutoComplete/AutocompleteLocationAndSalePoint';
import UserFormButtonGroup from '../UserFormButtonGroup';

jest.mock('@apiServices/usersService', () => ({
    editLocationAndSalePointUsers: jest.fn(),
    removeUser: jest.fn(),
    editLoginTypeUsers: jest.fn(),
}));

const mockHistoryPush = jest.fn();
let mockA = true;
const mockUsersArray = [usersTestArray[2], usersTestArray[3]];

jest.mock('react-router-dom', () => ({
    useParams: () => ({ userId: 1 }),
    useHistory: () => ({
        push: mockHistoryPush,
    }),
    useLocation: () => ({
        state: {
            users: mockA ? mockUsersArray : undefined,
        },
    }),
}));

const TEST_PROPS = {
    matchPath: 'TEST_PATH',
};

const TEST_USER_DATA = {
    LOCATION: testLocation,
    SALE_POINT: testSalePoint,
};

describe('<UserMultiEdit /> test', () => {
    const ComponentMount = () => mount(<UserMultiEdit {...TEST_PROPS} />);
    const ComponentShallow = () => shallow(<UserMultiEdit {...TEST_PROPS} />);

    it('should be mount snap', () => {
        const Component = ComponentMount();
        expect(Component.html()).toMatchSnapshot();
    });

    it('should find two users', () => {
        const Component = ComponentMount();
        expect(Component.find('.pageTitle').text()).toBe(`${CHOSE_USERS_COUNT} 2`);
    });

    it('should input edit users data and call submit function', async () => {
        const Component = ComponentShallow();
        const Autocomplete = Component.find(AutocompleteLocationAndSalePoint);
        Autocomplete.props().onLocationChange(TEST_USER_DATA.LOCATION);
        Autocomplete.props().onSalePointChange(TEST_USER_DATA.SALE_POINT);

        const ButtonGroup = Component.find(UserFormButtonGroup);
        ButtonGroup.props().onSubmit!();
        await sleep();
        expect(editLocationAndSalePointUsers).toBeCalledTimes(1);
        expect(mockHistoryPush).toBeCalledTimes(1);
    });

    it('should input edit users data and call submit function with catch', async () => {
        (editLocationAndSalePointUsers as jest.Mock).mockRejectedValue(new Error('Error'));
        const Component = ComponentShallow();
        const Autocomplete = Component.find(AutocompleteLocationAndSalePoint);
        Autocomplete.props().onLocationChange(TEST_USER_DATA.LOCATION);
        Autocomplete.props().onSalePointChange(TEST_USER_DATA.SALE_POINT);
        const ButtonGroup = Component.find(UserFormButtonGroup);
        ButtonGroup.props().onSubmit!();
        await sleep();
        expect(editLocationAndSalePointUsers).toBeCalledTimes(1);
        expect(ButtonGroup.find('.formError')).toBeTruthy();
    });

    it('should input edit users data and call submit function with error sale point', async () => {
        const Component = ComponentMount();
        const Autocomplete = Component.find(AutocompleteLocationAndSalePoint);

        act(() => {
            Autocomplete.props().onLocationChange(TEST_USER_DATA.LOCATION);
        });

        const ButtonGroup = Component.find(UserFormButtonGroup);

        act(() => {
            ButtonGroup.props().onSubmit!();
        });

        expect(Component.find('.formError')).toBeTruthy();
    });

    it('should call delete users function', async () => {
        const Component = ComponentMount();
        const ButtonGroup = Component.find(UserFormButtonGroup);

        await act(async () => {
            ButtonGroup.props().onDelete!();
        });

        expect(removeUser).toBeCalledTimes(2);
    });

    it('should call delete users function with catch', async () => {
        (removeUser as jest.Mock).mockRejectedValue(new Error('Error'));
        const Component = ComponentMount();
        const ButtonGroup = Component.find(UserFormButtonGroup);

        await act(async () => {
            ButtonGroup.props().onDelete!();
        });

        expect(ButtonGroup.find('.formError')).toBeTruthy();
    });

    it('should call cancel function', () => {
        const Component = ComponentShallow();
        const ButtonGroup = Component.find(UserFormButtonGroup);
        ButtonGroup.props().onCancel!();
        expect(mockHistoryPush).toBeCalledTimes(1);
    });

    it('should redirect to user page with users length equal 0', () => {
        mockA = false;

        act(() => {
            ComponentMount();
        });

        expect(mockHistoryPush).toBeCalledTimes(1);
        expect(mockHistoryPush).toBeCalledWith(TEST_PROPS.matchPath);
        mockA = true;
    });

});
