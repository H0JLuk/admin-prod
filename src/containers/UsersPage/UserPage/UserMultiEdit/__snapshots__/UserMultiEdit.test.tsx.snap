// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserMultiEdit /> test should be mount snap 1`] = `"<div class=\\"container\\"><div class=\\"backButton\\"><span role=\\"img\\" aria-label=\\"left\\" class=\\"anticon anticon-left\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"left\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z\\"></path></svg></span>  Назад</div><div class=\\"avatar\\"><span role=\\"img\\" aria-label=\\"user\\" style=\\"font-size: 22px; color: rgb(255, 255, 255); background-color: rgb(193, 200, 209); border-radius: 50px; width: 32px; height: 32px; padding: 3px;\\" tabindex=\\"-1\\" class=\\"anticon anticon-user ant-dropdown-trigger dropdown\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"user\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z\\"></path></svg></span></div></div><div class=\\"container\\"><div class=\\"header\\"><h4>Пользователи</h4><div class=\\"pageTitle\\">Выбрано пользователей 2</div></div><div class=\\"formContainer\\"><form class=\\"ant-form ant-form-horizontal ant-form-hide-required-mark form\\"><div class=\\"formColumn\\"><div class=\\"formLabel\\"><label for=\\"rc_select_0\\">Выберите локацию</label></div><div class=\\"inputWrapper\\"><div class=\\"ant-select ant-select-auto-complete autocompleteInput ant-select-single ant-select-allow-clear ant-select-customize-input ant-select-show-search\\"><div class=\\"ant-select-selector\\"><span class=\\"ant-select-selection-search\\"><span class=\\"ant-input-affix-wrapper ant-select-selection-search-input\\"><input autocomplete=\\"off\\" placeholder=\\"Область, город...\\" name=\\"location\\" type=\\"search\\" class=\\"ant-input\\" role=\\"combobox\\" aria-haspopup=\\"listbox\\" aria-owns=\\"rc_select_TEST_OR_SSR_list\\" aria-autocomplete=\\"list\\" aria-controls=\\"rc_select_TEST_OR_SSR_list\\" aria-activedescendant=\\"rc_select_TEST_OR_SSR_list_0\\" value=\\"\\" id=\\"rc_select_TEST_OR_SSR\\"><span class=\\"ant-input-suffix\\"><span role=\\"img\\" aria-label=\\"search\\" class=\\"anticon anticon-search suffix\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"search\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z\\"></path></svg></span></span></span></span><span class=\\"ant-select-selection-placeholder\\"></span></div></div></div></div><div class=\\"formColumn\\"><div class=\\"formLabel required\\"><label for=\\"rc_select_1\\">Выберите точку продажи</label></div><div><div class=\\"ant-select ant-select-auto-complete autocompleteInput ant-select-single ant-select-allow-clear ant-select-customize-input ant-select-show-search\\"><div class=\\"ant-select-selector\\"><span class=\\"ant-select-selection-search\\"><span class=\\"ant-input-affix-wrapper ant-select-selection-search-input\\"><input autocomplete=\\"off\\" placeholder=\\"Отделение ВСП\\" name=\\"salePoint\\" type=\\"search\\" class=\\"ant-input\\" role=\\"combobox\\" aria-haspopup=\\"listbox\\" aria-owns=\\"rc_select_TEST_OR_SSR_list\\" aria-autocomplete=\\"list\\" aria-controls=\\"rc_select_TEST_OR_SSR_list\\" aria-activedescendant=\\"rc_select_TEST_OR_SSR_list_0\\" value=\\"\\" id=\\"rc_select_TEST_OR_SSR\\"><span class=\\"ant-input-suffix\\"><span role=\\"img\\" aria-label=\\"search\\" class=\\"anticon anticon-search suffix\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"search\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z\\"></path></svg></span></span></span></span><span class=\\"ant-select-selection-placeholder\\"></span></div></div></div></div></form></div><div class=\\"buttonsContainer\\"><button type=\\"button\\" class=\\"ant-btn ant-btn-ghost\\"><span>Отменить</span></button><button type=\\"button\\" class=\\"ant-btn ant-btn-primary\\"><span>Сохранить</span></button><button type=\\"button\\" class=\\"ant-btn ant-btn-primary ant-btn-dangerous\\"><span>Удалить</span></button></div></div>"`;
